test/performance/performance-tests.sh

```diff
@@ -25,41 +26,237 @@
-initialize --num-nodes=4 --perf --cluster-version=1.25 "$@"
+set -o errexit
+set -o nounset
+set -o pipefail
 
-header "Running tests"
+declare JOB_NAME
+declare BUILD_ID
+declare ARTIFACTS
 
-function run_kperf() {
-  go_run knative.dev/kperf/cmd/kperf@latest "$@"
+ns="default"
+
+initialize --num-nodes=10 --cluster-version=1.25 "$@"
+
+env
+
+function scale_activator() {
+  local replicas=$1
+
+  echo "Setting activator replicas to ${replicas}"
+  kubectl -n "${SYSTEM_NAMESPACE}" patch hpa activator --patch "{\"spec\":{\"minReplicas\": ${replicas}, \"maxReplicas\": ${replicas} }}"
+
+  # Wait for HPA to do the scaling
+  sleep 30
 }
 
-mkdir -p "${ARTIFACTS}/kperf"
+function run_job() {
+  local name=$1
+  local file=$2
+
+  # cleanup from old runs
+  kubectl delete job "$name" -n "$ns" --ignore-not-found=true
+
+  # start the load test and get the logs
+  envsubst < "$file" | ko apply --sbom=none -Bf -
+
+  # sleep a bit to make sure the job is created
+  sleep 5
+
+  # Follow logs to wait for job termination
+  kubectl wait --for=condition=ready -n "$ns" pod --selector=job-name="$name" --timeout=-1s
+  kubectl logs -n "$ns" -f "job.batch/$name"
+
+  # Dump logs to a file to upload it as CI job artifact
+  kubectl logs -n "$ns" "job.batch/$name" >"$ARTIFACTS/$name.log"
+
+  # clean up
+  kubectl delete "job/$name" -n "$ns" --ignore-not-found=true
+  kubectl wait --for=delete "job/$name" --timeout=60s -n "$ns"
+}
+
+if ((IS_PROW)); then
+  export INFLUX_URL=$(cat /etc/influx-url-secret-volume/influxdb-url)
+  export INFLUX_TOKEN=$(cat /etc/influx-token-secret-volume/influxdb-token)
+else
+ export JOB_NAME="local"
+ export BUILD_ID="local"
+fi
+
+if [[ -z "${INFLUX_URL}" ]]; then
+  echo "env variable 'INFLUX_URL' not specified!"
+  exit 1
+fi
+if [[ -z "${INFLUX_TOKEN}" ]]; then
+  echo "env variable 'INFLUX_TOKEN' not specified!"
+  exit 1
+fi
+
+echo "Running load test with BUILD_ID: ${BUILD_ID}, JOB_NAME: ${JOB_NAME}, reporting results to: ${INFLUX_URL}"
+
+###############################################################################################
+header "Preparing cluster config"
+
+kubectl delete secret performance-test-config -n "$ns" --ignore-not-found=true
+kubectl create secret generic performance-test-config -n "$ns" \
+  --from-literal=influxurl="${INFLUX_URL}" \
+  --from-literal=influxtoken="${INFLUX_TOKEN}" \
+  --from-literal=jobname="${JOB_NAME}" \
+  --from-literal=buildid="${BUILD_ID}"
+
+# Tweak configuration for performance tests
+scale_activator 10
+toggle_feature rollout-duration 240 config-network
+toggle_feature scale-to-zero-grace-period 10s config-autoscaler
+toggle_feature kubernetes.podspec-init-containers enabled config-features # necessary for the real traffic test
+
+echo ">> Upload the test images"
+ko resolve --sbom=none -RBf test/test_images/autoscale > /dev/null
+ko resolve --sbom=none -RBf test/test_images/helloworld > /dev/null
+ko resolve --sbom=none -RBf test/test_images/slowstart > /dev/null
```

Comment:
This should be done already by `test_setup`

https://github.com/knative/serving/blob/32c2732efcf122ea9000145f9407ee138f9b9ca1/test/e2e-common.sh#L431-L432